include/maths/CModel.h

```diff
@@ -121,6 +121,11 @@ class MATHS_EXPORT CModelAddSamplesParams {
     //! Get the prior sample weights.
     const TDouble2VecWeightsAryVec& priorWeights() const;
 
+    //! Set the model change callback.
+    CModelAddSamplesParams& onModelChange(const maths_t::TModelChangeCallback& addAnnotation);
```

Comment:
In line with my previous comments, maybe something which captures that this is related to annotations, something like:
```suggestion
    CModelAddSamplesParams& addAnnotationCallback(const maths_t::TModelChangeCallback& addAnnotation);
```